zipline/data/data_portal.py

```diff
@@ -920,69 +950,49 @@ def get_splits(self, sids, dt):
 
         return splits
 
-    def get_dividends(self, sids, dt):
+    def get_stock_dividends(self, sid, trading_days):
```

Comment:
We should have `get_dividends` and `get_stock_dividends` logic implemented in the same place. Whether it is both in data_portal or both in adjustment_reader, my vote is for it to be in adjustment_reader, so that the read/write responsibility is located in one place.